packages/nextjs/src/utils/sdk.test.ts

```diff
@@ -74,6 +109,34 @@ describe("guessPotentiallyProxiedOrySdkUrl", () => {
     expect(guessPotentiallyProxiedOrySdkUrl()).toBe("https://myapp.vercel.app")
   })
 
+  it("should use NEXT_PUBLIC_VERCEL_URL when available", () => {
+    process.env["VERCEL_ENV"] = "preview"
+    process.env["NEXT_PUBLIC_VERCEL_URL"] = "public-myapp.vercel.app"
+    expect(guessPotentiallyProxiedOrySdkUrl()).toBe("https://public-myapp.vercel.app")
+  })
+
+  it("should use VERCEL_PROJECT_PRODUCTION_URL in production for vercel.app domains", () => {
```

Comment:
Is this actually true? If `VERCEL_ENV=production` we always use `ORY_SDK_URL`.

Reply:
The code was broken and the test was right, I fixed it so tests are passing now. If we run on a vercel.app domain, we have to use the NextJS middleware or otherwise cookies won't work.